api/projects.ts

```diff
@@ -1,9 +1,38 @@
 import { axios } from "./axios";
 import type { Project } from "./projects.types";
+import { ProjectStatus } from "@api/projects.types";
 
 const ENDPOINT = "/project";
 
 export async function getProjects() {
-  const { data } = await axios.get<Project[]>(ENDPOINT);
+  const { data } = await axios.get<Project[]>(ENDPOINT, {
+    transformResponse: [transformProjectsResponse],
+  });
+
   return data;
 }
+
+function transformProjectsResponse(data: string) {
```

Comment:
Instead of adding the data transformation here in the API layer I'd pass the response data as is to the UI. The data from the response represents the "real" project type. Then in the component you can use a function or a mapping object to transform the project status value to the required display value like this

```
function getStatusDisplayText(status: ProjectStatus) {
  switch(status) {
    case ProjectStatus.error:
      return "Critical";
    case ProjectStatus.warning:
      return "Warning";
    case ProjectStatus.info:
      return "Stable";
}
```

Then you can use this function in the JSX to render the text in the badge.

Reply:
Yeah, I hear you on that. My thinking was to munge the response to represent how the app would want to represent/use the `Project` datatype. Like, if we're hitting multiple apis that're represented by a Project type, yadda yadda yadda.

But I hear you! Changed! 👍